tests/webSocket-spec.coffee

```diff
@@ -0,0 +1,523 @@
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        ws.socket.addEventListener 'open', () ->
+          message = protocol.createMessage('/users')
+          ws.send(message, {})
+        spyOn(ws.callbacksBuffer, 'handle').and.callThrough()
+        ws.socket.addEventListener 'message', (e) ->
+          expect(ws.callbacksBuffer.handle).toHaveBeenCalledWith JSON.parse e.data
+          ws.disconnect()
+          done()
+
+    describe 'connection close', ->
+      it 'should fire "reconnecting" event when reconnect options is true', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        spyOn(ws, 'emit').and.callThrough()
+        ws.socket.addEventListener 'close', () ->
+          expect(ws.emit).toHaveBeenCalledWith 'reconnecting'
+          ws.disconnect()
+          done()
+        ws.socket.close()
+
+      it 'should reconnect when reconnect option is true', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        spyOn(ws, 'reconnect').and.callThrough()
+        ws.socket.addEventListener 'close', () ->
+          expect(ws.reconnect).toHaveBeenCalled()
+          ws.disconnect()
+          done()
+        ws.socket.close()
+
+      it 'should fire "close" event when reconnect options is false', (done) ->
+        newOptions =
+          reconnect: false
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', newOptions)
+        ws.connect()
+        spyOn(ws, 'emit').and.callThrough()
+        ws.socket.addEventListener 'close', () ->
+          expect(ws.emit).toHaveBeenCalledWith 'closed'
+          ws.disconnect()
+          done()
+        ws.socket.close()
+
+      it 'should disconnect when reconnect option is false', (done) ->
+        newOptions =
+          reconnect: false
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', newOptions)
+        ws.connect()
+        spyOn(ws, 'disconnect').and.callThrough()
+        ws.socket.addEventListener 'close', () ->
+          expect(ws.disconnect).toHaveBeenCalled()
+          ws.disconnect()
+          done()
+        ws.socket.close()
+
+  describe 'send', ->
+    describe 'channel connected', ->
+
+      it 'should send message with tag', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        spyOn(ws.socket, 'send').and.callThrough()
+        ws.socket.addEventListener 'open', () ->
+
+          ws.send(protocol.createMessage('/users'), {})
+          expect(ws.socket.send).toHaveBeenCalled()
+          ws.disconnect()
+          done()
+
+      it 'should buffer message callbacks', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        message = protocol.createMessage('/users')
+        {kind} = message
```

Comment:
I was trying to make `toHaveBeenCalledWith` but it didn't work  due to the callbacks